giraffe/src/components/geo/PointMapLayer.tsx

```diff
@@ -0,0 +1,201 @@
+// Libraries
+import React, {useEffect, useState} from 'react'
+import {FunctionComponent, useLayoutEffect} from 'react'
+import L from 'leaflet'
+import {Marker} from 'react-leaflet'
+import MarkerClusterGroup from 'react-leaflet-markercluster'
+import 'react-leaflet-markercluster/dist/styles.min.css'
+
+// Utils
+import {getColor} from './dimensionCalculations'
+import {SVGIcon} from './SVGIcon'
+import {defineToolTipEffect, formatValue} from './processing/toolTips'
+import {GeoTooltip} from './GeoTooltip'
+
+// Types
+import {GeoTable} from './processing/GeoTable'
+import {ClusterAggregation, GeoPointMapViewLayer} from '../../types/geo'
+import {Config} from '../../types'
+
+// Constants
+const MARKER_ICON_SIZE = [26, 38]
+const CLUSTER_ICON_SIZE = 30
+const CLUSTER_ICON_SIZE_BORDER = 4
+const DEFAULT_CLUSTER_COLOR = '#00a3ff'
+
+interface Props {
+  table: GeoTable
+  colorFieldName: string
+  properties: GeoPointMapViewLayer
+  stylingConfig: Partial<Config>
+  isClustered: boolean
+}
+
+const formatRowInfo = (
+  properties: GeoPointMapViewLayer,
+  table: GeoTable,
+  index
+) => {
+  const result = []
+  const timeValue = table.getTimeString(index)
+  if (timeValue)
+    result.push({
+      key: '_time',
+      name: 'Time',
+      type: 'string',
+      values: [timeValue],
+    })
+  const colorValue = table.getValue(index, properties.colorField)
+  const {colorDimension} = properties
+  const colorInfo = formatValue(
+    properties.colorField,
+    'Color',
+    colorValue,
+    colorDimension
+  )
+  if (colorInfo) result.push(colorInfo)
+  return result
+}
+
+const getClusterValues = (cluster, result: number[]): GeoPointMapViewLayer => {
+  let clusterRenderingProperties
+  for (const marker of cluster._markers) {
+    result.push(marker.value)
+    clusterRenderingProperties = marker.clusterRenderingProperties
+  }
+  for (const childCluster of cluster._childClusters) {
+    const properties = getClusterValues(childCluster, result)
+    if (properties) clusterRenderingProperties = properties
+  }
+  return clusterRenderingProperties
+}
+
+const getClusterValue = (
+  cluster
+): {value: number; properties: GeoPointMapViewLayer} => {
+  const values: number[] = []
+  const properties = getClusterValues(cluster, values)
+  const aggregation =
+    properties.clusterAggregationFunction || ClusterAggregation.max
+  switch (aggregation) {
+    case ClusterAggregation.max:
+      return {value: Math.max(...values), properties}
+    case ClusterAggregation.min:
+      return {value: Math.min(...values), properties}
+    case ClusterAggregation.mean:
+      return {
+        value: values.reduce((a, x) => a + x, 0) / values.length,
+        properties,
+      }
+    case ClusterAggregation.median:
+      if (values.length === 1) return {value: values[0], properties}
```

Comment:
Please always use braces for if statements. We will be updating the eslint rules to reflect this.